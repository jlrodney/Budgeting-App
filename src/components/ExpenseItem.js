import React from 'react';

const ExpenseItem = ({ id, description, amount, createdAt }) => (
  <div>
      <h3>{description}</h3>
    <p>Amount: Â£{amount/100} - Date: {createdAt}</p>
  </div>
);

export default ExpenseItem;
