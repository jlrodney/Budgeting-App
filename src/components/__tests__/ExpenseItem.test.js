import React from 'react';
import { shallow } from 'enzyme';
import ExpenseItem from '../ExpenseItem';
import moment from 'moment';


describe('Expense item', () => {
  it('should render correctly', () => {
    const expenses = [{
      id: '1',
      description: 'Elictricity',
      note: '',
      amount: 180,
      createdAt: 0
    }, {
      id: '2',
      description: 'Lunch',
      note: '',
      amount: 5,
      createdAt: moment(0).subtract(4, 'days').valueOf()
    }, {
      id: '3',
      description: 'Credit Card debt interest',
      note: '',
      amount: 60,
      createdAt: moment(0).add(4, 'days').valueOf()
    }];
    const wrapper = shallow(<ExpenseItem {...expenses[0]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
