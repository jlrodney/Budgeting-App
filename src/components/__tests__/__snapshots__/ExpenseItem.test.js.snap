// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expense item should render correctly 1`] = `
<div>
  <h3>
    Elictricity
  </h3>
  <p>
    Amount: Â£
    1.8
     - Date: 
    0
  </p>
</div>
`;
